linkerd/app/core/src/classify.rs

```diff
@@ -71,7 +71,18 @@ impl classify::Classify for Request {
         match self {
             Self::Profile(classes) => Response::Profile(classes.clone()),
 
-            Self::ClientPolicy(ClientPolicy::Http(policy)) => Response::Http(policy.clone()),
+            Self::ClientPolicy(ClientPolicy::Http(policy)) => {
+                let is_grpc = req
+                    .headers()
+                    .get(http::header::CONTENT_TYPE)
+                    .and_then(|v| v.to_str().ok())
+                    .map(|ct| ct.starts_with("application/grpc"))
+                    .unwrap_or(false);
+                if is_grpc {
+                    return Response::Grpc(Default::default());
+                }
```

Comment:
So, one difference between the behavior with `Request::Profile` and `Request::Default` is that in the `Profile` case, we will still use the configured HTTP classifier _if the response begins with a non-`200 OK` status code_ (which would indicate that it will not be terminated by a `grpc-status` (IIRC). In the `Default` case, we will always use the gRPC classifier (and just treat any 5xx HTTP status as an error), which is fine since we don't have a user-defined HTTP classifier. 

This code makes the `ClientPolicy::Http` classifier behave the same as the `Default` classifier, rather than the `Profile` classifier. I'm not sure if this is actually the most desirable behavior here --- would it be more correct to behave more like the `Profile` classifier, and use the configured HTTP classifier if the response has a 5xx HTTP status and there is no `grpc-status` trailer?

Reply:
If the `Content-Type: application/grpc` header is present, I would expect the request to be classified entirely based on `grpc-status` and for the HTTP response status to be ignored, even if it's a non-200.

Reply:
I also think that this behavior is pretty reasonable; I just wanted to point out that it's different from what we do for ServiceProfiles, and make sure we were okay with that.